modules/ServerAPI/src/client/SaltBrowser.coffee

```diff
@@ -0,0 +1,402 @@
+
+	render: =>
+		@template = _.template($('#SaltSummaryTableView').html())
+		$(@el).html @template
+		if @collection.models.length is 0
+			$(".bv_noMatchingSaltsFoundMessage").removeClass "hide"
+		else
+			$(".bv_noMatchingSaltsFoundMessage").addClass "hide"
+			@collection.each (salt) =>
+				prsc = new SaltRowSummaryController
+					model: salt
+				prsc.on "gotClick", @selectedRowChanged
+				@$("tbody").append prsc.render().el
+
+			@$("table").dataTable oLanguage:
+				sSearch: "Filter results: "
+		@
+
+
+class SaltBrowserController extends Backbone.View
+	events:
+		"click .bv_deleteSalt": "handleDeleteSaltClicked" 
+		"click .bv_confirmDeleteSaltButton": "handleConfirmDeleteSaltClicked"
+		"click .bv_cancelDelete": "handleCancelDeleteClicked"
+		"click .bv_downloadSaltBtn": "handleDownloadSaltClicked"
+		"click .bv_createNewSaltBtn": "handleCreateSaltClicked"
+		"click .bv_confirmCreateSaltButton": "handleConfirmCreateSaltClicked"
+		"click .bv_cancelCreate": "handleCancelCreateClicked"
+		"click .bv_editSalt": "handleEditSaltClicked"
+		"click .bv_confirmEditSaltButton": "handleConfirmEditSaltClicked"
+		"click .bv_cancelEdit":"handleCancelEditSaltClicked"
+
+	initialize: ->
+		template = _.template($("#SaltBrowserView").html())
+		$(@el).empty()
+		$(@el).html template
+		@searchController = new SaltSimpleSearchController
+			model: new SaltSearch()
+			el: @$('.bv_saltSearchController')
+		@searchController.render()
+		@searchController.on "searchReturned", @setupSaltSummaryTable.bind(@)
+		@$('.bv_queryToolDisplayName').html window.conf.service.result.viewer.displayName
+
+	setupSaltSummaryTable: (salts) =>
+		@destroySaltSummaryTable()
+
+		$(".bv_searchingSaltsMessage").addClass "hide"
+		if salts is null
+			@$(".bv_errorOccurredPerformingSearch").removeClass "hide"
+
+		else if salts.length is 0
+			@$(".bv_noMatchingSaltsFoundMessage").removeClass "hide"
+			@$(".bv_saltTableController").html ""
+		else
+			$(".bv_searchSaltsStatusIndicator").addClass "hide"
+			@$(".bv_saltTableController").removeClass "hide"
+			@saltSummaryTable = new SaltSummaryTableController
+				collection: new SaltList salts
+			@saltSummaryTable.on "selectedRowUpdated", @selectedSaltUpdated
+			$(".bv_saltTableController").html @saltSummaryTable.render().el
+
+	selectedSaltUpdated: (salt) =>
+		@trigger "selectedSaltUpdated"
+		@saltController = new SaltEditorController
+			model: new Salt salt.attributes
+			readOnly: true
+
+		$('.bv_saltController').html @saltController.render().el
+		$(".bv_saltController").removeClass("hide")
+		$(".bv_saltControllerContainer").removeClass("hide")
+
+		# Need to Use This Route (Or Similar) to Display Salt Structure
+		# app.get '/api/chemStructure/renderStructureByCode', exports.renderStructureByCode
+
+		# Need to Get 
+		requestJSON = {
+			#	"codeName" :  "#{@saltController.model.get("id")}"
+				"molStructure" : "#{@saltController.model.get("molStructure")}",
+				"height" : 100, 
```

Comment:
Not sure if anyone has any opinions on size or format

Reply:
Format looks good. I might make the size a little larger since we have plenty of real estate.